src/docfx/lib/HtmlUtility.cs

```diff
@@ -163,6 +164,52 @@ public static string TransformLinks(string html, Func<string, string> transform)
             return result.ToString();
         }
 
+        public static (List<(string uid, int line, int column)>, string) TransformXrefs(string html, int startLine, Func<string, (Error error, string href, string display, Document file)> transform)
+        {
+            var errors = new List<(string uid, int line, int column)>();
+
+            // Fast pass it does not have <xref> tag
+            if (!(html.Contains("<xref", StringComparison.OrdinalIgnoreCase) && html.Contains("href", StringComparison.OrdinalIgnoreCase)))
+            {
+                return (errors, html);
+            }
+
+            var doc = new HtmlDocument();
+            doc.LoadHtml(html);
+
+            var result = new StringBuilder(html.Length + 64);
+            var currentLine = startLine;
+            foreach (var node in doc.DocumentNode.Descendants())
+            {
+                if (node.Name != "xref")
+                {
+                    continue;
+                }
+
+                var xref = node.Attributes["href"];
+                if (xref is null)
+                {
+                    continue;
+                }
+
+                var rawSource = node.GetAttributeValue("data-raw-source", null);
+                var rawHtml = node.GetAttributeValue("data-raw-html", null);
+                var raw = HttpUtility.HtmlDecode(!string.IsNullOrEmpty(rawHtml) ? rawHtml : rawSource) ?? xref.Value;
+                var (_, resolvedHref, display, _) = transform(xref.Value);
+                if (string.IsNullOrEmpty(resolvedHref))
+                {
+                    errors.Add((xref.Value, currentLine, s_getValueStartIndex(xref)));
+                    result.Append($"@{raw}");
```

Comment:
Remove `@` here. `@` should be within `raw`.

Reply:
According to the test, seems not?